src/components/views/myDelegations/DelegateButtonModal.jsx

```diff
@@ -61,6 +63,25 @@ class DelegateButtonModal extends Component {
     this.selectedObject = this.selectedObject.bind(this);
   }
 
+  componentDidMount() {
```

Comment:
Please move the rate thing to only one hook, like `useCampaign`. The code is used three times

Reply:
Nice job @RamRamez